data_model/model.py

```diff
@@ -18,8 +18,9 @@ class UserRepository:
     def __init__(self):
         self.users = {}
 
-    def add_user(self, user_id):
-        self.users[user_id] = user_id
+    def add_user(self, user):
+        self.users[user.id] = user
+        self.users.update({'id': user})
```

Comment:
self.users should only  map id -> user objects. If you want to build the dictionary, just do that whenever a method requiring that dictionary is called.